File: Ruby/fib.rb
require 'benchmark'

def fib(n)
  return 0 if n == 0 || n == 1
  p, q = 0, 1
  if n > 0
    (n-1).times do
      p, q = q, p+q
    end
    return q
  else
    (-1-n).times do
      p, q = q-p, p
    end
    return q - p
  end
end

print "Enter the lower range of the Fibonacci series: "
lo = gets.chomp.to_i
print "Enter the higher range of the Fibonacci series: "
hi = gets.chomp.to_i

time = Benchmark.realtime do
  for i in lo..hi
    puts "#{i}: #{fib(i)}"
  end
end

puts time

File: C/fib.c
#include<stdio.h>

long double fib(int n) {
  long double p = 0, q = 1;
  if (n == 0)
    return 0;
  else if (n > 0) {
    for (int i = 1; i < n; i++) {
      q += p;
      p = q - p;
    }
    return q;
  } else
    for (int i = n; i < -1; i++) {
      p = q - p;
      q -= p;
    }
  return q - p;
}
    

int main()
{
  int m, n;
  //long double first = 0, second = 1, next;
  printf("Enter the lower range of the Fibonacci series: ");
  scanf("%i",&m);
  printf("Enter the higher range of the Fibonacci series: ");
  scanf("%i",&n);
 
  printf("The Fibonacci series between %i and %i are:\n", m, n);
 
  for (; m <= n ; m++)
    printf("%.0Lf\n",fib(m));

  return 0;
}
